ksqldb-engine/src/main/java/io/confluent/ksql/query/QueryBuilder.java

```diff
@@ -290,20 +311,30 @@ PersistentQueryMetadata buildPersistentQueryInDedicatedRuntime(
     final KeyFormat keyFormat;
     final ValueFormat valueFormat;
     final KsqlTopic ksqlTopic;
+    KeyFormat pullQueryKeyFormat;
```

Comment:
I still think you can just over write `keyFormat` directly as opposed to introducing `pullQueryKeyFormat` but maybe there's a reason they need to be separate?  It wasn't obvious from looking at the code.